bam_sort.c

```diff
@@ -3402,9 +3420,12 @@ int bam_sort_core_ext(SamOrder sam_order, char* sort_tag, int minimiser_kmer,
     // write sub files
     k = max_k = bam_mem_offset = 0;
     size_t name_len = strlen(prefix) + 30;
+    int placed = 0;
     while ((res = sam_read1(fp, header, b)) >= 0) {
         int mem_full = 0;
 
+        placed |= b->core.tid >= 0;
```

Comment:
Nice bit of conditional avoidance :smile:

Reply:
Lol yeah it's my usual coding style.   IOCCC learnt probably. ;-)